From: =?utf-8?q?Matti_Lehtim=C3=A4ki?= <matti.lehtimaki@gmail.com>
Date: Sat, 28 Oct 2023 23:52:23 +0300
Subject: media: dt-bindings: media: Add bindings for IMX800 sensor
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Add bindings documentaion for the Sony IMX800 camera sensor.

Signed-off-by: Matti Lehtimäki <matti.lehtimaki@gmail.com>
---
 .../i2c/{sony,imx412.yaml => sony,imx800.yaml}     | 54 +++++++++++++---------
 1 file changed, 33 insertions(+), 21 deletions(-)
 copy Documentation/devicetree/bindings/media/i2c/{sony,imx412.yaml => sony,imx800.yaml} (57%)

diff --git a/Documentation/devicetree/bindings/media/i2c/sony,imx412.yaml b/Documentation/devicetree/bindings/media/i2c/sony,imx800.yaml
similarity index 57%
copy from Documentation/devicetree/bindings/media/i2c/sony,imx412.yaml
copy to Documentation/devicetree/bindings/media/i2c/sony,imx800.yaml
index 5447ab0..d924b7c 100644
--- a/Documentation/devicetree/bindings/media/i2c/sony,imx412.yaml
+++ b/Documentation/devicetree/bindings/media/i2c/sony,imx800.yaml
@@ -1,43 +1,51 @@
 # SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
-# Copyright (C) 2021 Intel Corporation
 %YAML 1.2
 ---
-$id: http://devicetree.org/schemas/media/i2c/sony,imx412.yaml#
+$id: http://devicetree.org/schemas/media/i2c/sony,imx800.yaml#
 $schema: http://devicetree.org/meta-schemas/core.yaml#
 
-title: Sony IMX412 Sensor
+title: Sony IMX800 Sensor
 
 maintainers:
-  - Paul J. Murphy <paul.j.murphy@intel.com>
-  - Daniele Alessandrelli <daniele.alessandrelli@intel.com>
+  - Matti Lehtimäki <matti.lehtimaki@gmail.com>
 
-description:
-  IMX412 sensor is a Sony CMOS active pixel digital image sensor with an active
-  array size of 4072H x 3176V. It is programmable through I2C interface. The
+description: |-
+  IMX800 sensor is a Sony CMOS active pixel digital image sensor with an active
+  array size of 4096H x 3072V. It is programmable through I2C interface. The
   I2C client address is fixed to 0x1a as per sensor data sheet. Image data is
   sent through MIPI CSI-2.
 
 properties:
   compatible:
-    enum:
-      - sony,imx412
-      - sony,imx577
+    const: sony,imx800
+
   reg:
-    description: I2C address
+    description: I2C device address
     maxItems: 1
 
+  assigned-clocks: true
+  assigned-clock-parents: true
+  assigned-clock-rates: true
+
   clocks:
     description: Clock frequency 6MHz, 12MHz, 18MHz, 24MHz or 27MHz
     maxItems: 1
 
-  dovdd-supply:
-    description: Interface power supply.
+  vana1-supply:
+    description:
+      Analog voltage (VANA) supply, 2.8 V
 
-  avdd-supply:
-    description: Analog power supply.
+  vana2-supply:
+    description:
+      Analog voltage (VANA) supply, 1.8 V
 
-  dvdd-supply:
-    description: Digital power supply.
+  vdig-supply:
+    description:
+      Digital I/O voltage (VDIG) supply, 1.1 V
+
+  vif-supply:
+    description:
+      Interface voltage (VIF) supply, 1.2 or 1.8 V
 
   reset-gpios:
     description: Reference to the GPIO connected to the XCLR pin, if any.
@@ -78,12 +86,16 @@ examples:
         #size-cells = <0>;
 
         camera@1a {
-            compatible = "sony,imx412";
+            compatible = "sony,imx800";
             reg = <0x1a>;
-            clocks = <&imx412_clk>;
+            clocks = <&imx800_clk>;
+
+            assigned-clocks = <&imx800_clk>;
+            assigned-clock-parents = <&imx800_clk_parent>;
+            assigned-clock-rates = <24000000>;
 
             port {
-                imx412: endpoint {
+                imx800: endpoint {
                     remote-endpoint = <&cam>;
                     data-lanes = <1 2 3 4>;
                     link-frequencies = /bits/ 64 <600000000>;
